mods/weather/init.lua

```diff
@@ -114,22 +103,93 @@ local function update_clouds()
 			speed = {x = n_speedx * 4, z = n_speedz * 4},
 		})
 		-- now adjust the shadow intensity
-		player:set_lighting({ shadows = { intensity = 0.7 * (1 - density) } })
+		player:set_lighting({
+			shadows = { intensity = 0.7 * (1 - density) }
+		})
 	end
 end
 
+local function purge_effects(player)
+	-- reset potentially touched values to their defaults
+	if randomize_clouds then
+		player:set_clouds({
```

Comment:
Just pass no arguments, which resets the clouds to their default values.

Reply:
My idea here was to only reset values actually touched by the mod instead of the whole table. So if another mod would have changed the cloud height then that change doesn't get overridden needlessly. The drawback of this implementation is of course that it needs to get updated if the default values ever change.

Reply:
gah, why do the API docs never explicitly state whether absent values lead to overwriting with defaults or keeping the old values...